### PR TITLE
[WIP] Notification changes v2

### PR DESCRIPTION
**Pull request by edencakir**
TODO:
- [ ] [Made some native changes, but not working 🤔] Update app icon notification badge count
- [x] Disable OneSignal in-app notification
- [x] Mark as read when a notification is pressed
- [ ] [Big one, skipping] Don't show in-app notification if current screen is Notifications
- [ ] [Big one, solves the issue above as well] Consider replacing modal with rendered component
- [x] fix 99+ notifications (screenshot below)
- [x] tapping on mentions does nothing

simulator screen shot - iphone 6 - 2018-01-12 at 14 05 56

### CODE REVIEW

**Comment by JoshFabian** [comment]
@edencakir where we at with this PR?

**Comment by edencakir** [comment]
@JoshFabian Working on it!

**Comment by JoshFabian** [comment]
@edencakir @Mikunj should this be merged?

File: src/screens/Notifications/NotificationsScreen.js [hunk 1]
```diff
@@ -140,24 +162,44 @@ class NotificationsScreen extends PureComponent {
...
 
+  /**
+   * Fetches post by extracting postId from activity foreignId.
+   * Created for fetching mentions in a hacky way.
+   * @param {object} activity Activity object from notifications
+   * @returns {object} post
+   * @memberof NotificationsScreen
+   */
+  fetchPost = async (activity) => {
+    const postId = activity.foreignId.split(':')[1];
```
Comment (Mikunj) [inline src/screens/Notifications/NotificationsScreen.js]: Might want to check if activity.foreignId is not `null`

**Comment by edencakir** [comment]
@JoshFabian yep

**Comment by Mikunj** [comment]
💯

**Pull request merged**